packages/flame/lib/src/game/game_widget/game_widget.dart

```diff
@@ -198,16 +244,16 @@ class _GameWidgetState<T extends Game> extends State<GameWidget<T>> {
     if (oldWidget.game != widget.game) {
       // Reset the loaderFuture so that onMount will run again
       // (onLoad is still cached).
-      oldWidget.game.onRemove();
+      _currentGame.onRemove();
```

Comment:
Scenario 1:
When `oldWidget.game != null && widget.game != null && oldWidget.game != widget.game `
This should attribute` widget.game` to `_currentGame` after on remove. 

Scenario 2:
When `oldWidget.game != null && widget.game == null`
This should attribute` widget.create()` to `_currentGame` after on remove. 


Scenario 3:
When `oldWidget.game == null && widget.game != null `
This should attribute `widget.game` to `_currentGame` after on remove.

Reply:
Also, `removeGameStateListener` should be also called when removing a game instance from the widget.